presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoConfigUtils.java

```diff
@@ -74,4 +73,123 @@ public static void deployConfigFiles(File configTar, Path configDir)
+            else {
+                deleteDirectoryContents(configDir);
+                addWorkerConfig(configDir);
+            }
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to add config files", e);
+        }
+    }
+
+    private static void addEnvSh(Path configDir)
+            throws PrestoManagerException
+    {
+        Properties properties = new Properties();
+        properties.setProperty("JAVA8_HOME", System.getProperty("java.home"));
+        createPropertiesFile(properties, configDir.resolve("env.sh"), null);
+    }
+
+    private static void createPropertiesFile(Properties properties, Path path, String comments)
+            throws PrestoManagerException
+    {
+        try (OutputStream output = newOutputStream(path, CREATE)) {
+            properties.store(output, comments);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException(format("Failed to create properties file: %s", path.getFileName()), e);
+        }
+    }
+
+    public static void addConnectors(Path catalogDir)
+            throws PrestoManagerException
+    {
+        try {
+            if (isDirectory(DEFAULT_CATALOG_DIR)) {
+                copyDir(DEFAULT_CATALOG_DIR, catalogDir);
+            }
+            else {
+                addTpchCatalog(catalogDir);
+            }
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to add connectors", e);
+        }
+    }
+
+    private static void addTpchCatalog(Path catalogDir)
+            throws IOException
+    {
+        if (!isDirectory(catalogDir)) {
+            createDirectories(catalogDir);
+            write(catalogDir.resolve("tpch.properties"), "connector.name=tpch".getBytes());
+        }
+    }
+
+    private static void addWorkerConfig(Path configDir)
+            throws PrestoManagerException
+    {
+        addConfigProperties(configDir);
+        addNodeProperties(configDir);
+        addJvmConfig(configDir);
+        addEnvSh(configDir);
+    }
+
+    private static void addConfigProperties(Path configDir)
+            throws PrestoManagerException
+    {
+        // TODO: Add discovery.uri
+        Properties properties = new Properties();
+        properties.setProperty("coordinator", "false");
+        properties.setProperty("http-server.http.port", "8080");
+        properties.setProperty("query.max-memory", "50GB");
+        properties.setProperty("query.max-memory-per-node", "8GB");
+        createPropertiesFile(properties, configDir.resolve("config.properties"), "single node worker config");
+    }
+
+    private static void addNodeProperties(Path configDir)
+            throws PrestoManagerException
+    {
+        // TODO: Add node.server-log-file & node.launcher-log-file
+        Properties properties = new Properties();
+        properties.setProperty("node.environment", "presto");
+        properties.setProperty("node.id", randomUUID().toString());
```

Comment:
If the user specifies `preserveConfig` then node.id is preserved. If not, it'll be replaced.